backend/coreapp/views/scratch.py

```diff
@@ -427,6 +446,66 @@ def export(self, request: Request, pk):
             },
         )
 
+    @action(detail=True, methods=["POST"])
+    def pr(self, request: Request, pk):
+        scratch: Scratch = self.get_object()
+        if scratch.owner != request.profile:
+            raise ScratchNotOwnerException()
+        # TODO: catch exceptions
+        fn: Optional[ProjectFunction] = scratch.project_function
+        if fn is None:
+            raise ScratchNotProjectFunctionException()
+        user: Optional[User] = request.profile.user
+        if user is None:
+            raise GithubLoginException()
+        token = user.github.access_token
+        github_repo: Repository = fn.project.repo.details(token)
+
```

Comment:
some/all of this should be enforced by the frontend but I don't know how to do that

Reply:
I'll make a pass on the UI soon